blogs/comm-opt/README.md

```diff
@@ -0,0 +1,94 @@
+<div align="center">
+
+# Communictaion Optimization for Large-Scale Training 
+
+</div>
+
+
+## Table of Contents
+1. [Introduction](#introduction)
+2. [Parameter-update All-Reduce Optimization](#ar-opt)
+3. [ZeRO2 Gradient All-Gather Optimization](#ag-opt)
+4. [Sequence-Parallel Optimization](#sp-opt)
+5. [Sequence-Parallel Convergence Consideration](#convergence)
+
+
+## 1. Introduction <a name="introduction"></a>
+Training LLMs on large data-set can be costly both in terms of resource-budget and time-to-completion. An important step to minimize such cost is to find the right number of resources together with a scalable library that assures training under a time limit. In this post, we target a key component of the scalability feature at DeepSpeed, the communication optimization. There are several communication collectives used in the main DeepSpeed's technologies, which we go through their optimization in detail in the following sections.
+
+## 2. Parameter-update All-Reduce Optimization <a name="ar-opt"></a>
+
+Before diving into this techniques, let's take a step back and show some of the case studies that motivates this work of all-reduce optimization. 
+
+All-reduce operation is an important part of the the training process. In ZeRO, we handle this in buckets, which can be configured to get good comminication throughput. As the number of GPUs increases, we encounter smaller-partition all-reduces. In this case, the current bucketing scheme cannot help with the communication overhead. This mostly becomes an issue when training smaller-scale models (like Llama-7B) with large number of GPUs. 
+
+For instance, when training a dense-7B architecture with Zero stages 1 or 2, we encounter a 1 and 2 second increase for the all-reudce time by increasing from 256 to 512 and 1024 A100 GPUs. This issue mostly arises from the fact that, the gradient-averaging happens with smaller partitions (#parameters / #GPUs) per-GPU rank. This issue gets more serious when training MoE architectures (3 - 12 second) for which the expert's parameters can be farther away due to the current parallelism layout of data and expert parallelism.
```

Comment:
It would be better to present these results in table form showing the all-reduce time for different GPU count, and for dense and MoE models.

Reply:
I don't think there are too many data-points that make the text too verbose, however I am fine to add the table if you think this is the case.